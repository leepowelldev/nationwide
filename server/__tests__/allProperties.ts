import { gql } from 'apollo-server';
import {
  mockCollection,
  mockEmptyCollection,
} from './__utils__/mockCollections.js';
import setup from './__utils__/setup.js';

const query = gql`
  query {
    allProperties {
      id
      address
      type
      bedrooms
    }
  }
`;

describe('Query: allProperties', () => {
  test('should return array of properties', async () => {
    const { server } = setup(mockCollection);

    const res = await server.executeOperation({
      query,
    });

    expect(res).toMatchSnapshot();
  });

  test('should return empty array when no properties exist', async () => {
    const { server } = setup(mockEmptyCollection);

    const res = await server.executeOperation({
      query,
    });

    expect(res).toMatchSnapshot();
  });
});
