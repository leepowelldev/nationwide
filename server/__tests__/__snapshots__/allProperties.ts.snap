// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query: allProperties should return array of properties 1`] = `
Object {
  "data": Object {
    "allProperties": Array [
      Object {
        "address": "A test address",
        "bedrooms": 4,
        "id": "1",
        "type": "HOUSE",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Query: allProperties should return empty array when no properties exist 1`] = `
Object {
  "data": Object {
    "allProperties": Array [],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
